File: Java/JavaClient_v8/Junit_Appium/src/test/java/sampleTestAOSapp_iOS.java
import io.appium.java_client.AppiumBy;
import io.appium.java_client.ios.options.XCUITestOptions;
import io.appium.java_client.ios.IOSDriver;
import org.junit.*;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.net.URL;
import java.time.Duration;


public class sampleTestAOSapp_iOS {
    private IOSDriver driver;
    private String SERVER = "http://uftm-server:8080";

    @Before
    public void setUp() throws Exception {
        XCUITestOptions caps = new XCUITestOptions();
        caps.setCapability("platformName", "iOS");
        caps.setCapability("udid", "3ecdf8889a8b18cb8858410d1540f3e6b22a2cb1");
        caps.setCapability("bundleId", "com.mf.iShopping");
        caps.setCapability("uftm:oauthClientId", "oauth2-m2pbTT3wUdAvsdBJXr0w@microfocus.com");
        caps.setCapability("uftm:oauthClientSecret", "tcNG3V5YVdjbgJhPl3Vl");
        caps.setCapability("uftm:tenantId", "999999999");
        driver = new IOSDriver(new URL(SERVER + "/wd/hub"), caps);
    }

    @After
    public void teardown(){
        if (driver != null) {
            driver.quit();
        }
    }

    @Test
    public void testAOSapp(){
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.findElement(AppiumBy.xpath("//XCUIElementTypeStaticText[@name=\"LAPTOPS\"]")).click();
        driver.findElement(AppiumBy.accessibilityId("HP ENVY x360 - 15t Laptop")).click();
        wait.until(ExpectedConditions.presenceOfElementLocated(AppiumBy.accessibilityId("productDetailsButton"))).click();
        driver.findElement(AppiumBy.accessibilityId("Close2")).click();
        driver.findElement(AppiumBy.accessibilityId("quantityButtonId")).click();
        driver.findElement(AppiumBy.accessibilityId("Plus")).click();
        driver.findElement(AppiumBy.accessibilityId("APPLY")).click();
        driver.findElement(AppiumBy.accessibilityId("ADD TO CART")).click();
        driver.findElement(AppiumBy.accessibilityId("Ok")).click();
        driver.findElement(AppiumBy.accessibilityId("userNameLabel")).sendKeys("Username");
        driver.findElement(AppiumBy.accessibilityId("passwordLabel")).sendKeys("password");
        driver.findElement(AppiumBy.accessibilityId("LOGIN")).click();
        WebElement loginErrorText = wait.until(ExpectedConditions.presenceOfElementLocated(
                AppiumBy.accessibilityId("Incorrect user name or password.")));
        assert loginErrorText.isDisplayed() == true;
        wait.until(ExpectedConditions.presenceOfElementLocated(AppiumBy.accessibilityId("Menu"))).click();
        driver.findElement(AppiumBy.accessibilityId("HOME")).click();
    }
}

File: Python/iOS/sampleTestAOSapp_iOS.py
import unittest
from appium import webdriver
from appium.options.ios import XCUITestOptions
from appium.webdriver.common.appiumby import AppiumBy
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from datetime import timedelta

class AppiumTest(unittest.TestCase):    
    def setUp(self):
        options = XCUITestOptions().load_capabilities({                              
            "platformName": "iOS",
            "platformVersion": "15.1",
            'deviceName': 'iPhone 13 Mini',                                   
            "bundleId": "com.mf.iShopping",            
            # UFT Digital Lab capabilities:
            "uftm:oauthClientId": "oauth2-By6oMfOazCwqH6RE44xK@microfocus.com",
            "uftm:oauthClientSecret": "GDbTsTDf36qpPd03TlP2",
            "uftm:tenantId": "999999999",
            # disable SSL certificate verification if desired
            'verify': False 
        })       
        self.driver = webdriver.Remote('https://uftdl-server/wd/hub', options=options)

    def test_app(self):
        wait = WebDriverWait(self.driver, timeout=timedelta(seconds=10).total_seconds())
        self.driver.find_element(AppiumBy.XPATH, "//XCUIElementTypeStaticText[@name=\"LAPTOPS\"]").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "HP ENVY x360 - 15t Laptop").click()
        wait.until(EC.presence_of_element_located((AppiumBy.ACCESSIBILITY_ID, "productDetailsButton"))).click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "Close2").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "quantityButtonId").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "Plus").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "APPLY").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "ADD TO CART").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "Ok").click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "userNameLabel").send_keys("Username")
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "passwordLabel").send_keys("password")
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "LOGIN").click()
        login_error_text = wait.until(EC.presence_of_element_located(
            (AppiumBy.ACCESSIBILITY_ID, "Incorrect user name or password.")
        ))
        assert login_error_text.is_displayed() == True
        wait.until(EC.presence_of_element_located((AppiumBy.ACCESSIBILITY_ID, "Menu"))).click()
        self.driver.find_element(AppiumBy.ACCESSIBILITY_ID, "HOME").click()

    def tearDown(self):
        if self.driver is not None:
            self.driver.quit()

if __name__ == '__main__':
    suite = unittest.TestLoader().loadTestsFromTestCase(AppiumTest)
    unittest.TextTestRunner(verbosity=2).run(suite)